### PR TITLE
 dotnet/templating#2994: adds missing localization for error messages in TemplateEngine.Edge

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
fixes dotnet/templating#2994 

### Solution
Failure messages returned by Edge were not localized, the PR adds localization for them.
Note: some of the messages are not localizable at the moment: https://github.com/dotnet/templating/issues/3194

### Checks:
- [x] Added unit tests - already available
- [x] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx [hunk 1]
```diff
@@ -167,7 +179,55 @@
...
+  <data name="TemplateCreator_TemplateCreationResult_Error_CouldNotLoadTemplate" xml:space="preserve">
+    <value>Could not load template.</value>
+  </data>
+  <data name="TemplateCreator_TemplateCreationResult_Error_CreationFailed" xml:space="preserve">
+    <value>Failed to create template, details: {0}.</value>
+  </data>
+  <data name="TemplateCreator_TemplateCreationResult_Error_DestructiveChanges" xml:space="preserve">
+    <value>Destructive changes detected.</value>
+  </data>
+  <data name="TemplateCreator_TemplateCreationResult_Error_InvalidParameterValues" xml:space="preserve">
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx:227]: Not used + emtpy?

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx [hunk 2]
```diff
@@ -167,7 +179,55 @@
...
+    <value />
+  </data>
+  <data name="TemplateCreator_TemplateCreationResult_Error_MissingRequiredParameters" xml:space="preserve">
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx:230]: Not used + emtpy?

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.Designer.cs [hunk 3]
```diff
@@ -212,5 +350,50 @@ internal class LocalizableStrings {
...
+        internal static string TemplateCreator_TemplateCreationResult_Error_CreationFailed {
+            get {
+                return ResourceManager.GetString("TemplateCreator_TemplateCreationResult_Error_CreationFailed", resourceCulture);
+            }
+        }
+        
+        /// <summary>
+        ///   Looks up a localized string similar to Destructive changes detected..
+        /// </summary>
+        internal static string TemplateCreator_TemplateCreationResult_Error_DestructiveChanges {
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.Edge/LocalizableStrings.Designer.cs:375]: Where is this used?

**Review by DavidKarlas — CHANGES_REQUESTED** [review]
I'm a bit conflicted about localizing `catch(Exception e)` code path, since that means localizers will have to translate something that might not ever be used...

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx (hunk 1 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx:227]: Removed

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx (hunk 2 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx:230]: Removed

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.Designer.cs (hunk 3 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Edge/LocalizableStrings.Designer.cs:375]: fixed, used in TemplateCreation.InstantiateAsynx

**Review by DavidKarlas — APPROVED** [review]
See above comment about SettingsLoader_Error_FailedToScan

**Comment by DavidKarlas** [comment]
Ideally once https://github.com/dotnet/roslyn-analyzers/issues/4865 is fixed we can catch this easier...

File: src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetInstaller.cs [hunk 4]
```diff
@@ -238,17 +293,23 @@ public Task<UninstallResult> UninstallAsync(IManagedTemplatePackage templatePack
...
             {
                 _environmentSettings.Host.FileSystem.FileDelete(templatePackage.MountPointUri);
                 return Task.FromResult(UninstallResult.CreateSuccess(templatePackage));
             }
-            catch (Exception ex)
+            catch (Exception e)
             {
-                _logger.LogDebug($"Uninstalling {templatePackage.DisplayName} failed. Details:{ex.ToString()}");
-                return Task.FromResult(UninstallResult.CreateFailure(templatePackage, InstallerErrorCode.GenericError, $"Failed to uninstall {templatePackage.DisplayName}, reason: {ex.Message}"));
+                _logger.LogDebug($"Uninstalling {templatePackage.DisplayName} failed. Details:{e.ToString()}");
```
Comment (bekir-ozturk) [inline src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetInstaller.cs:308]: Not your change, but wanted to add. I think it is better performance-wise to leave the formatting to the `ILogger` like below. If "Debug" level is too low and the message won't be displayed, then the final string won't be constructed.
```suggestion
                _logger.LogDebug($"Uninstalling {0} failed. Details: {1}", templatePackage.DisplayName, e.ToString());
```

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx [hunk 5]
```diff
@@ -125,6 +128,15 @@
     <value>version {0}</value>
     <comment>small letters, string is used in the sentence</comment>
   </data>
+  <data name="GlobalSettingsTemplatePackageProvider_InstallResult_Error_MultipleInstallersCanBeUsed" xml:space="preserve">
+    <value>{0} can be installed by several installers, specify the installer name to use.</value>
```
Comment (bekir-ozturk) [inline src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx:132]:
```suggestion
    <value>{0} can be installed by several installers. Specify the installer name to be used.</value>
```

**Review by bekir-ozturk — APPROVED** [review]
LGTM. Some minor suggestions.

File: src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx [hunk 6]
```diff
@@ -167,7 +179,55 @@
   <data name="NuGetInstaller_Error_FileAlreadyExists" xml:space="preserve">
     <value>File {0} already exists</value>
   </data>
+  <data name="NuGetInstaller_InstallResut_Error_DownloadFailed" xml:space="preserve">
+    <value>Failed to download {0} from {1}.</value>
+  </data>
+  <data name="NuGetInstaller_InstallResut_Error_InstallGeneric" xml:space="preserve">
+    <value>Failed to install the package {0}, details: {1}.</value>
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.Edge/LocalizableStrings.resx]: I put details from new line here, and below as ex.ToString() usually produce multiline output.

**Pull request merged**